py3status/request.py

```diff
@@ -63,6 +63,7 @@ def __init__(self, url, params, data, headers, timeout, auth):
             elif isinstance(e, HTTPError):
                 self._status_code = e.code
                 self._error_message = reason
+                self._response = e
```

Comment:
I included this since I read the response when OWM gives a failure.

```python
req = self.py3.request(url, timeout=self.request_timeout)
if(req.status_code != 200):
    data = req.json()
    raise OWMException(data['message'])
```

According to the [docs](https://docs.python.org/2/library/urllib2.html#urllib2.HTTPError), the error object acts as a response-like object so you can interpret the content of the page.

The above would not work without the change here, since the `req.json()` would fail, since `self._response` would be undefined.

I can remove it, but it gives richer error messages to the end user. I could always assume that the error is API Key related, since that's usually the case, and it would remove the need for the change.

Thoughts?